### DISCUSSION TITLE
system reliability, flexibility and stability

### DISCUSSION THREAD

**Discussion by wanyingw193** [NREL/ReEDS-2.0#108] [Q&A: Features]
Hi ReEDS team, I was wondering how ReEDS handle system reliability, flexibility and stability. Thanks!

**Comment by wesleyjcole** [comment]
Can you be a bit more specific here?  Each of this is a large topic in and of itself, so it's difficult to provide a useful answer.  The ReEDS model documentation will be a helpful starting point if you haven't already looked at those pieces there.

**Reply by wanyingw193** [reply]
Hello! Thank you for your reply. I went through the model documentation, but I feel like I didn't capture all the topics. So I was just trying to get some information from this question.

In my understanding, reliability in the model is handled by capacity credit, scheduled reserve ratio and operating reserve. Are there other components that cover reliability?
Then, flexibility is covered by demand response, which allows the user to enter different EFS electrification levels, and auxiliary reserve requirements (operating reserves). Is this the case?
But I didn't find the description of system stability. Could I have some guidance from you? Thanks!

**Comment by wesleyjcole** [comment]
Reliability is captured via the planning reserve margin and the operating reserve requirements.  That extends into transmission trading and the Augur algorithms that calculate capacity credit at an hourly basis.

Flexibility is represented in the demand response, the operating reserves (particularly the "flex" product), the minimum generation and ramping constraints, and transmission treatment.

I'm not sure what you mean by system stability.  Things like voltage stability are not represented in ReEDS, but some of the operating reserve products (e.g., regulation reserves) might cover pieces of what you are looking for.

**Comment by wanyingw193** [comment]
Hi! If I want to calculate the capacity to be used for planning reserves only (not include peak demand) and I use the following formula, does it make sense? Thanks!

Planning reserve capacity = total capacity (CAP.l) * generator availability (fraction) - generation capacity (GEN.l) - operating capacity (OPRES.l).

Btw, when calculating generator availability, why does the outage rate not vary depending on location, model year and technical class V? Why are there no planned outages in summer? Thank you!

**Reply by wanyingw193** [reply]
Hi, this is a update on above question. 

Is it possible to calculate how much of the planning reserve margin capacity is allocated to each type of technology? The equation may look like the following?
Planning reserve margin capacity (MW) = Firm capacity (MW) - peak demand (MW)?

Because generation capacity is used to meet the demand and supply balance (eq_supply_demand_balance). Could the above equation update to
Planning reserve margin capacity(MW) = Firm capacity(MW) - Generation capacity(MW)?

Thanks!

**Reply by wesleyjcole** [reply]
The cap_firm parameter outputs the firm capacity contribution of each technology in each region.  That parameter is output in the cap_firm.csv file, so you can take a look at it there.

The planning reserve constraint is separate from the generation constraint.  Planning reserves are for meeting the required planning reserve margin in the planning timescale.  They are designed to ensure that you have enough capacity to be able to operate the grid reliably on a day-to-day basis.  Once you get into the operational timescale (where the eq_supply_demand_balance comes into play), then planning reserves don't matter because you can't build any new resources within that timescale.  At that point you are just dispatching what you have.

In ReEDS we simply solve the operational timescale and planning timescale at the same time.  Capacity that you build to meet the planning reserve margin constraint is then available at the operational timescale for dispatch.

**Comment by wanyingw193** [comment]
Hi ReEDS team,

For the eq_reserve_margin(r,szn,t) constraint, is the left side representing firm capacity and the right side representing peak demand and planning reserve margin?

If the above understanding is correct, why is it necessary to constrain 'the firm capacity =g= peak demand *(1+prm)' instead of using the =e= symbol?

Is it reasonable to set prm(r,t)=0 in this constraint?

Thank you!

screenshot:
![image](https://user-images.githubusercontent.com/78582696/192946416-cec7b7ae-007f-4378-96bf-bbeb9f52afb8.png)
![image](https://user-images.githubusercontent.com/78582696/192946480-686290e0-626f-4ade-814f-e59c4e0c82e1.png)

**Reply by wesleyjcole** [reply]
Your understanding is correct.  Grid entitities often exceed the NERC recommended planning reserve margin levels.  See, for example, our work that summarizes this at https://www.sciencedirect.com/science/article/abs/pii/S0301421518306797.  This paper also explore what happens when you implement higher reserve margins in ReEDS.

If you set the prm to zero, then ReEDS will build out to the peak and will not have any excess capacity.  That means that if you were to run a dispatch model with outages and load uncertainty you would see quite a bit of dropped load.

**Reply by wanyingw193** [reply]
Thank you!
